kombu/transport/pubsub.py

```diff
@@ -247,28 +249,26 @@ def exchange_declare(self, exchange='', **kwargs):
         :param exchange: queue name
         :type body: str
         """
-        to_add = False
         if exchange not in self.state.exchanges:
             logger.info("".join(["Topic: ", exchange, " not found added in state"]))
             topic_path = self._get_topic_path(exchange)
-            try:
-                logger.info("Creating new topic: " + exchange)
-                self.publisher.create_topic(topic_path)
-                to_add = True
-            except AlreadyExists:
-                to_add = True
-            except Exception as e:
-                raise ChannelError(
-                    '{0} - no exchange {1!r} in vhost {2!r}'.format(
-                        e.__str__(),
-                        exchange,
-                        self.connection.client.virtual_host or '/'),
-                    (50, 10), 'Channel.exchange_declare', '404',
-                )
-            finally:
-                logger.info("".join(["adding topic: ", exchange, " to state"]))
-                if to_add:
-                    self.state.exchanges[exchange] = topic_path
+            if not self.topics_map.get(exchange, False):
```

Comment:
this is again creating topics during runtime, lets create/validate everything on initialisation phase only

Reply:
@mpsmks this will only create the topic if the name of the topic is not passed in the `transport options` .. we just need to make sure that a valid map of topics is passed when starting the celery app/workers

Reply:
so if we pass this as false for all the topics it will not create the topics at all right ?

but we need topics to be created during new env creation